layers/common_layers.py

```diff
@@ -105,14 +105,100 @@ def forward(self, attention_cat):
         return processed_attention
 
 
-class Attention(nn.Module):
+class GravesAttention(nn.Module):
+    """ Graves attention as described here:
+        - https://arxiv.org/abs/1910.10288
+    """
+    COEF = 0.3989422917366028  # numpy.sqrt(1/(2*numpy.pi))
+
+    def __init__(self, query_dim, K):
+        super(GravesAttention, self).__init__()
+        self._mask_value = 0.0
+        self.K = K
+        # self.attention_alignment = 0.05
+        self.eps = 1e-5
+        self.J = None
+        self.N_a = nn.Sequential(
+            nn.Linear(query_dim, query_dim, bias=True),
+            nn.ReLU(),
+            nn.Linear(query_dim, 3*K, bias=True))
+        self.attention_weights = None
+        self.mu_prev = None
+        self.init_layers()
+
+    def init_layers(self):
+        torch.nn.init.constant_(self.N_a[2].bias[10:15], 0.5)
+        torch.nn.init.constant_(self.N_a[2].bias[5:10], 10)
+
+    def init_states(self, inputs):
+        if self.J is None or inputs.shape[1] > self.J.shape[-1]:
+            self.J = torch.arange(0, inputs.shape[1]).to(inputs.device).expand([inputs.shape[0], self.K, inputs.shape[1]])
```

Comment:
I tried out this branch out of curiosity, but I had a dtype mismatch in `forward` due to `J` being a Long. I'm really not familiar with the code, but would `torch.arange(0.0` be a reasonable fix?

Reply:
which version pytorch?

Reply:
Pytorch 1.1.0 (on python 3.7.5)

Reply:
try the latest pytorch, I guess the behaviour has been changed. But yes you can also solve as you said.